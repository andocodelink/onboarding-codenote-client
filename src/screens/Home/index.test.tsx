import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';

import { Home, mapStateToProps, mapDispatchToProps } from './index';

describe('Home screen', () => {
  const userHasAuthenticated = () => ({});
  const jdomAlert = window.alert;

  beforeAll(() => {
    window.alert = jest.fn();
  });

  afterAll(() => {
    window.alert = jdomAlert;
  })
  
  test('renders without authenication', () => {
    const wrapper = shallow(<Home isAuthenticated={false} userHasAuthenticated={userHasAuthenticated} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders with authenication', () => {
    const wrapper = shallow(<Home isAuthenticated userHasAuthenticated={userHasAuthenticated} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('validate states with 10 notes', async () => {
    let windowAlert: any = window.alert;
    windowAlert.mockClear();
    const wrapper = mount(<MemoryRouter><Home isAuthenticated  userHasAuthenticated={userHasAuthenticated} /></MemoryRouter>);
    await Promise.resolve();
    expect(wrapper.find('Home').state().notes).toHaveLength(10);
  });

  test('map state to props', () => {
    let props = mapStateToProps({
      authenticate: {
        isAuthenticated: true 
      }
    });
    expect(props.isAuthenticated).toBe(true);
  });

  test('map dispatch to props', () => {
    let dispatch = jest.fn();
    let props =  mapDispatchToProps(dispatch);
    expect(props.userHasAuthenticated).toBeInstanceOf(Function);
  });

})