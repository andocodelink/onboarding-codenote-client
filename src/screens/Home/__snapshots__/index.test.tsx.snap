// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen renders with authenication 1`] = `
<div
  className="Home"
>
  <div
    className="notes"
  >
    <PageHeader
      bsClass="page-header"
    >
      All notes
    </PageHeader>
    <ListGroup
      bsClass="list-group"
    />
  </div>
</div>
`;

exports[`Home screen renders without authenication 1`] = `
<div
  className="Home"
>
  <div
    className="lander"
  >
    <h1>
      CodeNote
    </h1>
    <p>
      A simple note taking app by me
    </p>
  </div>
</div>
`;
